src/input-processing/inputProcessing.js

```diff
@@ -263,24 +266,54 @@ function replaceSpecialLetters(input) {
-        8: /^(((((opcao)|(numero))\s)?(oito|8(\.0)?))|(oitav(a|o)(\sopcao)?))$/,
-        9: /^(((((opcao)|(numero))\s)?(nove|9(\.0)?))|(non(a|o)(\sopcao)?))$/,
-        10: /^(((((opcao)|(numero))\s)?(dez|10(\.0)?))|(decima(a|o)(\sopcao)?))$/,
-        11: /^(((((opcao)|(numero))\s)?(onze|11(\.0)?))|(decim(a|o)\s(primeir(a|o))(\sopcao)?))$/,
-        12: /^(((((opcao)|(numero))\s)?(doze|12(\.0)?))|(decim(a|o)\s(segund(a|o))(\sopcao)?))$/,
-        13: /^(((((opcao)|(numero))\s)?(treze|13(\.0)?))|(decim(a|o)\s(terceir(a|o))(\sopcao)?))$/,
-        14: /^(((((opcao)|(numero))\s)?(quartorze|14(\.0)?))|(decim(a|o)\s(quart(a|o))(\sopcao)?))$/,
-        15: /^(((((opcao)|(numero))\s)?(quinze|15(\.0)?))|(decim(a|o)\s(quint(a|o))(\sopcao)?))$/
+        1: {
+            'en-US': /^(((((option)|(number))\s)?(one|1(\.0)?))|(first(\soption)?))$/,
+            'pt-BR': /^(((((opcao)|(numero))\s)?(um|1(\.0)?))|(primeir(a|o)(\sopcao)?))$/,
+        },
+        2: {
+            'en-US': /^(((((option)|(number))\s)?(two|2(\.0)?))|(second(\soption)?))$/,
+            'pt-BR': /^(((((opcao)|(numero))\s)?(dois|2(\.0)?))|(segund(a|o)(\sopcao)?))$/,
+        },
+        3: {
+            'en-US': /^(((((option)|(number))\s)?(three|3(\.0)?))|(third(\soption)?))$/,
+            'pt-BR': /^(((((opcao)|(numero))\s)?(tres|3(\.0)?))|(terceir(a|o)(\sopcao)?))$/,
+        },
+        4: {
+            'en-US': /^(((((option)|(number))\s)?(four|4(\.0)?))|(fourth(\soption)?))$/,
+            'pt-BR': /^(((((opcao)|(numero))\s)?(quatro|4(\.0)?))|(quart(a|o)(\sopcao)?))$/,
+        },
+        5: {
+            'en-US': /^(((((option)|(number))\s)?(five|5(\.0)?))|(fifth(\soption)?))$/,
+            'pt-BR': /^(((((opcao)|(numero))\s)?(cinco|5(\.0)?))|(quint(a|o)(\sopcao)?))$/,
+        },
+        6: {
+            'en-US': /^(((((option)|(number))\s)?(six|6(\.0)?))|(sixth(\soption)?))$/,
+            'pt-BR': /^(((((opcao)|(numero))\s)?(seis|6(\.0)?))|(sext(a|o)(\sopcao)?))$/,
+        },
+        7: {
+            'en-US': /^(((((option)|(number))\s)?(seven|7(\.0)?))|(seventh(\soption)?))$/,
+            'pt-BR': /^(((((opcao)|(numero))\s)?(sete|7(\.0)?))|(setim(a|o)(\sopcao)?))$/,
+        },
+        8: {
+            'en-US': /^(((((option)|(number))\s)?(eight|8(\.0)?))|(eighth(\soption)?))$/,
+            'pt-BR': /^(((((opcao)|(numero))\s)?(oito|8(\.0)?))|(oitav(a|o)(\sopcao)?))$/,
+        },
+        9: {
+            'en-US': /^(((((option)|(number))\s)?(nine|9(\.0)?))|(nineth(\soption)?))$/,
+            'pt-BR': /^(((((opcao)|(numero))\s)?(nove|9(\.0)?))|(non(a|o)(\sopcao)?))$/,
+        },
+        10: {
+            'en-US': /^(((((option)|(number))\s)?(ten|10(\.0)?))|(tenth(\soption)?))$/,
+            'pt-BR': /^(((((opcao)|(numero))\s)?(dez|10(\.0)?))|(decim(a|o)(\sopcao)?))$/,
+        }
     };
+
+    if(userLanguage){
+        return numbersWrittenRegex[`${number}`][userLanguage];
+    }
+
     return numbersWrittenRegex[`${number}`];
```

Comment:
That I remember, in this case he is considering that the user is passing options in a structure like this:

```
 const numbersWrittenRegex = {
        1: /^(((((opcao)|(numero))\s)?(um|1(\.0)?))|(primeir(a|o)(\sopcao)?))$/,
        2: /^(((((opcao)|(numero))\s)?(dois|2(\.0)?))|(segund(a|o)(\sopcao)?))$/,
//...
}
```

In both cases the return is a string.

This way, this script is still compatible with the latest version. But I agree it can be confusing. I think this should have a good documentation to eliminate one possible confusion.